### PR TITLE
OAuth 2.0 new SDK feature documentation

### PR DESCRIPTION
**Pull request by MichaelGHSeg**
### Proposed changes

Adding OAuth information to Node and Python

### Merge timing

Depending on OAuth release

### CODE REVIEW

File: src/connections/sources/catalog/libraries/server/node/index.md [hunk 1]
```diff
@@ -635,6 +635,38 @@ const analytics = new Analytics({
 })
 ```
 
+### OAuth 2
+
+In order to guarantee authorized communication between your server environment and Segment's Tracking API, you can enable OAuth 2 in your Segment workspace.  To support the non-interactive server environment, the OAuth workflow used is a signed client assertion JWT.  You will need a public and private key pair where the public key is uploaded to the segment dashboard and the private key is kept in your server environment to be used by this SDK. Your server will verify its identity by signing a token request and will receive a token that is used to to authorize all communication with the Segment Tracking API.
```
Comment (rchinn-segment) [inline src/connections/sources/catalog/libraries/server/node/index.md:640]:
```suggestion
Enable OAuth 2 in your Segment workspace to guarantee authorized communication between your server environment and Segment's Tracking API. To support the non-interactive server environment, the OAuth workflow used is a signed client assertion JWT.  

 You will need a public and private key pair where:
 - The public key is uploaded to the Segment dashboard.
 - The private key is kept in your server environment to be used by this SDK.
 Your server will verify its identity by signing a token request, and will receive a token that is used to authorize all communication with the Segment Tracking API.
```

File: src/connections/sources/catalog/libraries/server/node/index.md [hunk 2]
```diff
@@ -635,6 +635,38 @@ const analytics = new Analytics({
...
+
+You will also need to provide the OAuth Application ID and the public key's ID, both of which are provided in the Segment dashboard.  There are also options available to specify the authorization server, custom scope, maximum number of retries, or a custom HTTP client if your environment has special rules for separate segment endpoints.
```
Comment (rchinn-segment) [inline src/connections/sources/catalog/libraries/server/node/index.md:642]:
```suggestion
You'll need to provide the OAuth Application ID and the public key's ID, both of which are provided in the Segment dashboard.  There are also options available to specify the authorization server, custom scope, maximum number of retries, or a custom HTTP client if your environment has special rules for separate segment endpoints.
```

File: src/connections/sources/catalog/libraries/server/node/index.md [hunk 3]
```diff
@@ -635,6 +635,38 @@ const analytics = new Analytics({
...
+
+You should ensure that you are implementing handling for Analytics SDK errors.  Good logging will help distinguish any configuration issues.
```
Comment (rchinn-segment) [inline src/connections/sources/catalog/libraries/server/node/index.md:644]:
```suggestion
Be sure to implement handling for Analytics SDK errors. Good logging will help distinguish any configuration issues.
```

File: src/connections/sources/catalog/libraries/server/node/index.md [hunk 4]
```diff
@@ -635,6 +635,38 @@ const analytics = new Analytics({
...
+
+For more information, see the [Segment OAuth 2.0 documentation](/docs/connections/oauth/)
```
Comment (rchinn-segment) [inline src/connections/sources/catalog/libraries/server/node/index.md:646]:
```suggestion
For more information, see the [Segment OAuth 2.0 documentation](/docs/connections/oauth/).
```

File: src/connections/sources/catalog/libraries/server/python/index.md [hunk 5]
```diff
@@ -560,6 +560,35 @@ analytics.track('9742', 'Song Played', context={
 
 Be sure to see the full [reference of supported keys](/docs/connections/spec/common/#context).
 
+### OAuth 2
+
+In order to guarantee authorized communication between your server environment and Segment's Tracking API, you can enable OAuth 2 in your Segment workspace.  To support the non-interactive server environment, the OAuth workflow used is a signed client assertion JWT.  You will need a public and private key pair where the public key is uploaded to the segment dashboard and the private key is kept in your server environment to be used by this SDK. Your server will verify its identity by signing a token request and will receive a token that is used to to authorize all communication with the Segment Tracking API.
```
Comment (rchinn-segment) [inline src/connections/sources/catalog/libraries/server/python/index.md:565]:
```suggestion
Enable OAuth 2 in your Segment workspace to guarantee authorized communication between your server environment and Segment's Tracking API. To support the non-interactive server environment, the OAuth workflow used is a signed client assertion JWT.  

You will need a public and private key pair where:
- The public key is uploaded to the Segment dashboard. 
- The private key is kept in your server environment to be used by this SDK. 
Your server will verify its identity by signing a token request and will receive a token that is used to authorize all communication with the Segment Tracking API.
```

File: src/connections/sources/catalog/libraries/server/python/index.md [hunk 6]
```diff
@@ -560,6 +560,35 @@ analytics.track('9742', 'Song Played', context={
...
+
+You will also need to provide the OAuth Application ID and the public key's ID, both of which are provided in the Segment dashboard.  There are also options available to specify the authorization server, custom scope, maximum number of retries, or a custom HTTP client if your environment has special rules for separate segment endpoints.
```
Comment (rchinn-segment) [inline src/connections/sources/catalog/libraries/server/python/index.md:567]:
```suggestion
You'll need to provide the OAuth Application ID and the public key's ID, both of which are provided in the Segment dashboard.  There are also options available to specify the authorization server, custom scope, maximum number of retries, or a custom HTTP client if your environment has special rules for separate segment endpoints.
```

File: src/connections/sources/catalog/libraries/server/python/index.md [hunk 7]
```diff
@@ -560,6 +560,35 @@ analytics.track('9742', 'Song Played', context={
...
+
+You should ensure that you are implementing handling for Analytics SDK errors.  Good logging will help distinguish any configuration issues.
```
Comment (rchinn-segment) [inline src/connections/sources/catalog/libraries/server/python/index.md:569]:
```suggestion
Be sure to implement handling for Analytics SDK errors. Good logging will help distinguish any configuration issues.
```

File: src/connections/sources/catalog/libraries/server/python/index.md [hunk 8]
```diff
@@ -560,6 +560,35 @@ analytics.track('9742', 'Song Played', context={
...
+
+For more information, see the [Segment OAuth 2.0 documentation](/docs/connections/oauth/)
```
Comment (rchinn-segment) [inline src/connections/sources/catalog/libraries/server/python/index.md:571]:
```suggestion
For more information, see the [Segment OAuth 2.0 documentation](/docs/connections/oauth/).
```

**Review by rchinn-segment — CHANGES_REQUESTED** [review]
Thanks for submitting this, @MichaelGHSeg. Just a few formatting suggestions, but other than that looks good!

File: src/connections/sources/catalog/libraries/server/node/index.md [hunk 9]
```diff
@@ -635,6 +635,38 @@ const analytics = new Analytics({
 })
 ```
 
+### OAuth 2
```
Comment (stayseesong) [inline src/connections/sources/catalog/libraries/server/node/index.md:638]:
```suggestion
### OAuth 2.0
```

Comment (stayseesong) [reply src/connections/sources/catalog/libraries/server/node/index.md:638]: @rchinn-segment all mentions of OAuth 2 should be OAuth 2.0 as that's what we call it in other parts of the docs site. Think it'd be good if you add a hyperlink somewhere to the OAuth 2.0 page https://segment.com/docs/connections/oauth/

File: src/connections/sources/catalog/libraries/server/node/index.md [hunk 10]
```diff
@@ -635,6 +635,38 @@ const analytics = new Analytics({
 })
 ```
 
+### OAuth 2.0
+
+In order to guarantee authorized communication between your server environment and Segment's Tracking API, you can enable [OAuth 2.0](/docs/connections/oauth/) in your Segment workspace.  To support the non-interactive server environment, the OAuth workflow used is a signed client assertion JWT.  You will need a public and private key pair where the public key is uploaded to the segment dashboard and the private key is kept in your server environment to be used by this SDK. Your server will verify its identity by signing a token request and will receive a token that is used to to authorize all communication with the Segment Tracking API.
```
Comment (rchinn-segment) [inline src/connections/sources/catalog/libraries/server/node/index.md:640]:
```suggestion
Enable [OAuth 2](/docs/connections/oauth/) in your Segment workspace to guarantee authorized communication between your server environment and Segment's Tracking API. To support the non-interactive server environment, the OAuth workflow used is a signed client assertion JWT.  

You will need a public and private key pair where:
- The public key is uploaded to the Segment dashboard. 
- The private key is kept in your server environment to be used by this SDK. 
Your server will verify its identity by signing a token request and will receive a token that is used to to authorize all communication with the Segment Tracking API.
```

File: src/connections/sources/catalog/libraries/server/python/index.md [hunk 11]
```diff
@@ -560,6 +560,35 @@ analytics.track('9742', 'Song Played', context={
 
 Be sure to see the full [reference of supported keys](/docs/connections/spec/common/#context).
 
+### OAuth 2.0
+
+In order to guarantee authorized communication between your server environment and Segment's Tracking API, you can enable [OAuth 2.0](/docs/connections/oauth/) in your Segment workspace.  To support the non-interactive server environment, the OAuth workflow used is a signed client assertion JWT.  You will need a public and private key pair where the public key is uploaded to the segment dashboard and the private key is kept in your server environment to be used by this SDK. Your server will verify its identity by signing a token request and will receive a token that is used to to authorize all communication with the Segment Tracking API.
```
Comment (rchinn-segment) [inline src/connections/sources/catalog/libraries/server/python/index.md:565]:
```suggestion
Enable [OAuth 2](/docs/connections/oauth/) in your Segment workspace to guarantee authorized communication between your server environment and Segment's Tracking API. To support the non-interactive server environment, the OAuth workflow used is a signed client assertion JWT.  

You will need a public and private key pair where:
- The public key is uploaded to the Segment dashboard. 
- The private key is kept in your server environment to be used by this SDK. 
Your server will verify its identity by signing a token request and will receive a token that is used to to authorize all communication with the Segment Tracking API.
```

**Review by rchinn-segment — APPROVED** [review]

File: src/connections/sources/catalog/libraries/server/python/index.md [hunk 12]
```diff
@@ -560,6 +560,35 @@ analytics.track('9742', 'Song Played', context={
 
 Be sure to see the full [reference of supported keys](/docs/connections/spec/common/#context).
 
+### OAuth 2
```
Comment (stayseesong) [inline src/connections/sources/catalog/libraries/server/python/index.md]:
```suggestion
## OAuth 2
```

File: src/connections/sources/catalog/libraries/server/node/index.md (hunk 9 above)
Comment (stayseesong) [inline src/connections/sources/catalog/libraries/server/node/index.md]:
```suggestion
## OAuth 2
```

**Review by stayseesong — CHANGES_REQUESTED** [review]
I just realized that OAuth 2.0 should be in its own section and not nested under "Troubleshooting" or "Multiple Clients"

**Comment by stayseesong** [comment]
@rchinn-segment I'll take care of merging this PR in as it'll go out with the OAuth docs that I'm releasing later this week

**Review by stayseesong — APPROVED** [review]

**Pull request merged**